### PR TITLE
Improve multithreading of mmaps_generator

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Improve multithreading of mmaps_generator by letting threads handle single tiles instead of whole maps
- Add various new CMake options to support different *San sanitizer tools when using Clang

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**

None

**Tests performed:**

- Tested map 30 with 4 threads, the result looks good in RecastDemo

**Known issues and TODO list:** (add/remove lines as needed)

- [x] Run *San tests
- [ ] Check ingame that mmaps still work
- [x] Fix mmaps_generator not closing when processing 1 single tile

### CODE REVIEW

File: src/tools/mmaps_generator/MapBuilder.cpp [hunk 1]
```diff
@@ -176,40 +180,46 @@ namespace MMAP
...
 
             if (_cancelationToken)
                 return;
 
-            buildMap(mapId);
+            dtNavMesh* navMesh = dtAllocNavMesh();
+            if (!navMesh->init(&tileInfo.m_navMeshParams))
+            {
+                printf("[Map %03i] Failed creating navmesh for tile %i,%i !\n", tileInfo.m_mapId, tileInfo.m_tileX, tileInfo.m_tileY);
+                return;
```
Comment (Shauren) [inline src/tools/mmaps_generator/MapBuilder.cpp:194]: leaks navmesh if init fails

**Comment by Faq** [comment]
`**Store TileBuilder as raw pointer for now, it will be changed later on** `
When?

**Comment by jackpoz** [comment]
> `Store TileBuilder as raw pointer for now, it will be changed later on`
> Out of this PR?

It depends, first I want to make sure it works and there are no race conditions. There are a few ToDo left and some are about polishing, like moving TileBuilder class to its own file, so I would say modern C++ refactoring would fit better in another PR

**Comment by jackpoz** [comment]
This PR is now complete from my side

**Pull request merged**